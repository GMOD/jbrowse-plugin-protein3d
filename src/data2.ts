export const feature={
  "strand": -1,
  "source": "HAVANA",
  "type": "transcript",
  "start": 51287257,
  "end": 51297990,
  "phase": 0,
  "derived_features": [],
  "refName": "chr1",
  "id": "ENST00000530004.5",
  "parent": "ENSG00000085831.16",
  "gene_id": "ENSG00000085831.16",
  "transcript_id": "ENST00000530004.5",
  "gene_type": "protein_coding",
  "gene_name": "TTC39A",
  "transcript_type": "protein_coding",
  "transcript_name": "TTC39A-216",
  "level": "2",
  "protein_id": "ENSP00000431228.1",
  "transcript_support_level": "1",
  "hgnc_id": "HGNC:18657",
  "tag": [
    "basic",
    "CCDS"
  ],
  "ccdsid": "CCDS76162.1",
  "havana_gene": "OTTHUMG00000008193.5",
  "havana_transcript": "OTTHUMT00000387467.1",
  "subfeatures": [
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51287257,
      "end": 51288280,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:7",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "7",
      "exon_id": "ENSE00000835623.5",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-0",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "three_prime_UTR",
      "start": 51287257,
      "end": 51288156,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "UTR3:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "7",
      "exon_id": "ENSE00000835623.5",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-1",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "CDS",
      "start": 51288156,
      "end": 51288280,
      "phase": 1,
      "derived_features": [],
      "refName": "chr1",
      "id": "CDS:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "7",
      "exon_id": "ENSE00000835623.5",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-2",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "CDS",
      "start": 51288838,
      "end": 51288955,
      "phase": 1,
      "derived_features": [],
      "refName": "chr1",
      "id": "CDS:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "6",
      "exon_id": "ENSE00003633626.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-3",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "CDS",
      "start": 51290004,
      "end": 51290119,
      "phase": 2,
      "derived_features": [],
      "refName": "chr1",
      "id": "CDS:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "5",
      "exon_id": "ENSE00003508941.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-4",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "CDS",
      "start": 51290513,
      "end": 51290625,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "CDS:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "4",
      "exon_id": "ENSE00003479848.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-5",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "CDS",
      "start": 51294390,
      "end": 51294511,
      "phase": 1,
      "derived_features": [],
      "refName": "chr1",
      "id": "CDS:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "3",
      "exon_id": "ENSE00003479532.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-6",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "CDS",
      "start": 51296078,
      "end": 51296155,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "CDS:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "2",
      "exon_id": "ENSE00003608742.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-7",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "stop_codon",
      "start": 51288156,
      "end": 51288159,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "stop_codon:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "7",
      "exon_id": "ENSE00000835623.5",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-8",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51288838,
      "end": 51288955,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:6",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "6",
      "exon_id": "ENSE00003633626.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-9",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51290004,
      "end": 51290119,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "5",
      "exon_id": "ENSE00003508941.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-10",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51290513,
      "end": 51290625,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:4",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "4",
      "exon_id": "ENSE00003479848.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-11",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51294390,
      "end": 51294511,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:3",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "3",
      "exon_id": "ENSE00003479532.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-12",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51296078,
      "end": 51296170,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:2",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "2",
      "exon_id": "ENSE00003608742.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-13",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "start_codon",
      "start": 51296152,
      "end": 51296155,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "start_codon:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "2",
      "exon_id": "ENSE00003608742.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-14",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "five_prime_UTR",
      "start": 51296155,
      "end": 51296170,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "UTR5:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "2",
      "exon_id": "ENSE00003608742.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-15",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "five_prime_UTR",
      "start": 51297790,
      "end": 51297990,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "UTR5:ENST00000530004.5",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "1",
      "exon_id": "ENSE00002148799.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-16",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    },
    {
      "strand": -1,
      "source": "HAVANA",
      "type": "exon",
      "start": 51297790,
      "end": 51297990,
      "phase": 0,
      "derived_features": [],
      "refName": "chr1",
      "id": "exon:ENST00000530004.5:1",
      "parent": "ENST00000530004.5",
      "gene_id": "ENSG00000085831.16",
      "transcript_id": "ENST00000530004.5",
      "gene_type": "protein_coding",
      "gene_name": "TTC39A",
      "transcript_type": "protein_coding",
      "transcript_name": "TTC39A-216",
      "exon_number": "1",
      "exon_id": "ENSE00002148799.1",
      "level": "2",
      "protein_id": "ENSP00000431228.1",
      "transcript_support_level": "1",
      "hgnc_id": "HGNC:18657",
      "tag": [
        "basic",
        "CCDS"
      ],
      "ccdsid": "CCDS76162.1",
      "havana_gene": "OTTHUMG00000008193.5",
      "havana_transcript": "OTTHUMT00000387467.1",
      "uniqueId": "590611027-offset-601794870-0-17",
      "parentId": "590611027-offset-601794870-0",
      "__jbrowsefmt": {}
    }
  ],
  "uniqueId": "590611027-offset-601794870-0",
  "parentId": "590611027-offset-601794870",
  "__jbrowsefmt": {}
}

  export const alignment={
    consensus:
      '                                                                                                                                                                                                                                                                                                                                                                                                        ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ',
    alns: [
      {
        id: 'a',
        seq: 'MGQKGHKDSLYPCGGTPESSLHEALDQCMTALDLFLTNQFSEALSYLKPRTKESMYHSLTYATILEMQAMMTFDPQDILLAGNMMKEAQMLCQRHRRKSSVTDSFSSLVNRPTLGQFTEEEIHAEVCYAECLLQRAALTFLQGSSHGGAVRPRALHDPSHACSCPPGPGRQHLFLLQDENMVSFIKGGIKVRNSYQTYKELDSLVQSSQYCKGENHPHFEGGVKLGVGAFNLTLSMLPTRILRLLEFVGFSGNKDYGLLQLEEGASGHSFRSVLCVMLLLCYHTFLTFVLGTGNVNIEEAEKLLKPYLNRYPKGAIFLFFAGRIEVIKGNIDAAIRRFEECCEAQQHWKQFHHMCYWELMWCFTYKGQWKMSYFYADLLSKENCWSKATYIYMKAAYLSMFGKEDHKPFGDDEVELFRAVPGLKLKIAGKSLPTEKFAIRKSRRYFSSNPISLPVPALEMMYIWNGYAVIGKQPKLTDGILEIITKAEEMLEKGPENEYSVDDECLVKLLKGLCLKYLGRVQEAEENFRSISANEKKIKYDHYLIPNALLELALLLMEQDRNEEAIKLLESAKQNYKNYSMESRTHFRIQAATLQAKSSLENSSRSMVSSVSL-',
      },
      {
        id: 'b',
        seq: '--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------MKAAYLSMFGKEDHKPFGDDEVELFRAVPGLKLKIAGKSLPTEKFAIRKSRRYFSSNPISLPVPALEMMYIWNGYAVIGKQPKLTDGILEIITKAEEMLEKGPENEYSVDDECLVKLLKGLCLKYLGRVQEAEENFRSISANEKKIKYDHYLIPNALLELALLLMEQDRNEEAIKLLESAKQNYKNYSMESRTHFRIQAATLQAKSSLENSSRSMVSSVSL*',
      },
    ],
  }
