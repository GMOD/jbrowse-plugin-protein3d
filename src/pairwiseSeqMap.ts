export interface Alignment {
  alns: {
    id: string
    seq: string
  }[]
}

export function pairwiseSeqMap(alignment: Alignment) {
  const k1 = alignment.alns[0].seq
  const k2 = alignment.alns[1].seq
  if (k1.length !== k2.length) {
    throw new Error('mismatched length')
  }

  let j = 0
  let k = 0
  const coord1 = {} as Record<string, number | undefined>
  const coord2 = {} as Record<string, number | undefined>
  const mismatch1 = {} as Record<string, boolean | undefined>
  const mismatch2 = {} as Record<string, boolean | undefined>

  // eslint-disable-next-line unicorn/no-for-loop
  for (let i = k; i < k1.length; i++) {
    const char1 = k1[i]
    const char2 = k2[i]

    if (char1 === char2) {
      coord1[j] = k
      coord2[k] = j
      k++
      j++
    } else if (char2 === '-') {
      j++
    } else if (char1 === '-') {
      k++
    } else {
      coord1[j] = k
      coord2[k] = j
      mismatch1[j] = true
      mismatch2[k] = true
      k++
      j++
    }
  }
  return { coord1, coord2, mismatch1, mismatch2 }
}
