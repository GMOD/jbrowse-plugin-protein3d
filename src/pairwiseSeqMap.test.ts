import { pairwiseSeqMap } from './pairwiseSeqMap'
import { alignment } from './data2'
test('test', () => {
  const ret = pairwiseSeqMap(alignment)
  expect(ret).toMatchSnapshot()
})
