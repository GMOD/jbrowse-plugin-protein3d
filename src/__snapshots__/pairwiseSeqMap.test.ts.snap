// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
{
  "coord1": {
    "392": 0,
    "393": 1,
    "394": 2,
    "395": 3,
    "396": 4,
    "397": 5,
    "398": 6,
    "399": 7,
    "400": 8,
    "401": 9,
    "402": 10,
    "403": 11,
    "404": 12,
    "405": 13,
    "406": 14,
    "407": 15,
    "408": 16,
    "409": 17,
    "410": 18,
    "411": 19,
    "412": 20,
    "413": 21,
    "414": 22,
    "415": 23,
    "416": 24,
    "417": 25,
    "418": 26,
    "419": 27,
    "420": 28,
    "421": 29,
    "422": 30,
    "423": 31,
    "424": 32,
    "425": 33,
    "426": 34,
    "427": 35,
    "428": 36,
    "429": 37,
    "430": 38,
    "431": 39,
    "432": 40,
    "433": 41,
    "434": 42,
    "435": 43,
    "436": 44,
    "437": 45,
    "438": 46,
    "439": 47,
    "440": 48,
    "441": 49,
    "442": 50,
    "443": 51,
    "444": 52,
    "445": 53,
    "446": 54,
    "447": 55,
    "448": 56,
    "449": 57,
    "450": 58,
    "451": 59,
    "452": 60,
    "453": 61,
    "454": 62,
    "455": 63,
    "456": 64,
    "457": 65,
    "458": 66,
    "459": 67,
    "460": 68,
    "461": 69,
    "462": 70,
    "463": 71,
    "464": 72,
    "465": 73,
    "466": 74,
    "467": 75,
    "468": 76,
    "469": 77,
    "470": 78,
    "471": 79,
    "472": 80,
    "473": 81,
    "474": 82,
    "475": 83,
    "476": 84,
    "477": 85,
    "478": 86,
    "479": 87,
    "480": 88,
    "481": 89,
    "482": 90,
    "483": 91,
    "484": 92,
    "485": 93,
    "486": 94,
    "487": 95,
    "488": 96,
    "489": 97,
    "490": 98,
    "491": 99,
    "492": 100,
    "493": 101,
    "494": 102,
    "495": 103,
    "496": 104,
    "497": 105,
    "498": 106,
    "499": 107,
    "500": 108,
    "501": 109,
    "502": 110,
    "503": 111,
    "504": 112,
    "505": 113,
    "506": 114,
    "507": 115,
    "508": 116,
    "509": 117,
    "510": 118,
    "511": 119,
    "512": 120,
    "513": 121,
    "514": 122,
    "515": 123,
    "516": 124,
    "517": 125,
    "518": 126,
    "519": 127,
    "520": 128,
    "521": 129,
    "522": 130,
    "523": 131,
    "524": 132,
    "525": 133,
    "526": 134,
    "527": 135,
    "528": 136,
    "529": 137,
    "530": 138,
    "531": 139,
    "532": 140,
    "533": 141,
    "534": 142,
    "535": 143,
    "536": 144,
    "537": 145,
    "538": 146,
    "539": 147,
    "540": 148,
    "541": 149,
    "542": 150,
    "543": 151,
    "544": 152,
    "545": 153,
    "546": 154,
    "547": 155,
    "548": 156,
    "549": 157,
    "550": 158,
    "551": 159,
    "552": 160,
    "553": 161,
    "554": 162,
    "555": 163,
    "556": 164,
    "557": 165,
    "558": 166,
    "559": 167,
    "560": 168,
    "561": 169,
    "562": 170,
    "563": 171,
    "564": 172,
    "565": 173,
    "566": 174,
    "567": 175,
    "568": 176,
    "569": 177,
    "570": 178,
    "571": 179,
    "572": 180,
    "573": 181,
    "574": 182,
    "575": 183,
    "576": 184,
    "577": 185,
    "578": 186,
    "579": 187,
    "580": 188,
    "581": 189,
    "582": 190,
    "583": 191,
    "584": 192,
    "585": 193,
    "586": 194,
    "587": 195,
    "588": 196,
    "589": 197,
    "590": 198,
    "591": 199,
    "592": 200,
    "593": 201,
    "594": 202,
    "595": 203,
    "596": 204,
    "597": 205,
    "598": 206,
    "599": 207,
    "600": 208,
    "601": 209,
    "602": 210,
    "603": 211,
    "604": 212,
    "605": 213,
    "606": 214,
    "607": 215,
    "608": 216,
    "609": 217,
    "610": 218,
    "611": 219,
    "612": 220,
  },
  "coord2": {
    "0": 392,
    "1": 393,
    "10": 402,
    "100": 492,
    "101": 493,
    "102": 494,
    "103": 495,
    "104": 496,
    "105": 497,
    "106": 498,
    "107": 499,
    "108": 500,
    "109": 501,
    "11": 403,
    "110": 502,
    "111": 503,
    "112": 504,
    "113": 505,
    "114": 506,
    "115": 507,
    "116": 508,
    "117": 509,
    "118": 510,
    "119": 511,
    "12": 404,
    "120": 512,
    "121": 513,
    "122": 514,
    "123": 515,
    "124": 516,
    "125": 517,
    "126": 518,
    "127": 519,
    "128": 520,
    "129": 521,
    "13": 405,
    "130": 522,
    "131": 523,
    "132": 524,
    "133": 525,
    "134": 526,
    "135": 527,
    "136": 528,
    "137": 529,
    "138": 530,
    "139": 531,
    "14": 406,
    "140": 532,
    "141": 533,
    "142": 534,
    "143": 535,
    "144": 536,
    "145": 537,
    "146": 538,
    "147": 539,
    "148": 540,
    "149": 541,
    "15": 407,
    "150": 542,
    "151": 543,
    "152": 544,
    "153": 545,
    "154": 546,
    "155": 547,
    "156": 548,
    "157": 549,
    "158": 550,
    "159": 551,
    "16": 408,
    "160": 552,
    "161": 553,
    "162": 554,
    "163": 555,
    "164": 556,
    "165": 557,
    "166": 558,
    "167": 559,
    "168": 560,
    "169": 561,
    "17": 409,
    "170": 562,
    "171": 563,
    "172": 564,
    "173": 565,
    "174": 566,
    "175": 567,
    "176": 568,
    "177": 569,
    "178": 570,
    "179": 571,
    "18": 410,
    "180": 572,
    "181": 573,
    "182": 574,
    "183": 575,
    "184": 576,
    "185": 577,
    "186": 578,
    "187": 579,
    "188": 580,
    "189": 581,
    "19": 411,
    "190": 582,
    "191": 583,
    "192": 584,
    "193": 585,
    "194": 586,
    "195": 587,
    "196": 588,
    "197": 589,
    "198": 590,
    "199": 591,
    "2": 394,
    "20": 412,
    "200": 592,
    "201": 593,
    "202": 594,
    "203": 595,
    "204": 596,
    "205": 597,
    "206": 598,
    "207": 599,
    "208": 600,
    "209": 601,
    "21": 413,
    "210": 602,
    "211": 603,
    "212": 604,
    "213": 605,
    "214": 606,
    "215": 607,
    "216": 608,
    "217": 609,
    "218": 610,
    "219": 611,
    "22": 414,
    "220": 612,
    "23": 415,
    "24": 416,
    "25": 417,
    "26": 418,
    "27": 419,
    "28": 420,
    "29": 421,
    "3": 395,
    "30": 422,
    "31": 423,
    "32": 424,
    "33": 425,
    "34": 426,
    "35": 427,
    "36": 428,
    "37": 429,
    "38": 430,
    "39": 431,
    "4": 396,
    "40": 432,
    "41": 433,
    "42": 434,
    "43": 435,
    "44": 436,
    "45": 437,
    "46": 438,
    "47": 439,
    "48": 440,
    "49": 441,
    "5": 397,
    "50": 442,
    "51": 443,
    "52": 444,
    "53": 445,
    "54": 446,
    "55": 447,
    "56": 448,
    "57": 449,
    "58": 450,
    "59": 451,
    "6": 398,
    "60": 452,
    "61": 453,
    "62": 454,
    "63": 455,
    "64": 456,
    "65": 457,
    "66": 458,
    "67": 459,
    "68": 460,
    "69": 461,
    "7": 399,
    "70": 462,
    "71": 463,
    "72": 464,
    "73": 465,
    "74": 466,
    "75": 467,
    "76": 468,
    "77": 469,
    "78": 470,
    "79": 471,
    "8": 400,
    "80": 472,
    "81": 473,
    "82": 474,
    "83": 475,
    "84": 476,
    "85": 477,
    "86": 478,
    "87": 479,
    "88": 480,
    "89": 481,
    "9": 401,
    "90": 482,
    "91": 483,
    "92": 484,
    "93": 485,
    "94": 486,
    "95": 487,
    "96": 488,
    "97": 489,
    "98": 490,
    "99": 491,
  },
  "mismatch1": {},
  "mismatch2": {},
}
`;
